frontend/src/operator/webui/src/app/device.service.ts

```diff
@@ -1,37 +1,59 @@
 import {Injectable} from '@angular/core';
-import {map, mergeMap, shareReplay} from 'rxjs/operators';
-import {Observable, Subject, merge} from 'rxjs';
+import {combineLatestWith, map, mergeMap, shareReplay} from 'rxjs/operators';
+import {Observable, Subject, BehaviorSubject} from 'rxjs';
 import {HttpClient} from '@angular/common/http';
-import {DeviceInfo} from './device-info-interface';
 import {DomSanitizer} from '@angular/platform-browser';
+import {DeviceInfo} from './device-info-interface';
+import {DeviceItem} from './device-item.interface';
 
 @Injectable({
   providedIn: 'root',
 })
 export class DeviceService {
-  private refreshSubject = new Subject<void>();
-  private deviceFromServer = this.httpClient
-    .get<string[]>('./devices')
-    .pipe(
-      map((deviceIds: string[]) =>
-        deviceIds.sort((a, b) => a.localeCompare(b, undefined, {numeric: true}))
-      )
-    );
+  private refreshSubject = new BehaviorSubject<void>(undefined);
+
+  private groupIdSubject = new Subject<string | null>();
 
-  private devices = merge(
-    this.deviceFromServer,
-    this.refreshSubject.pipe(mergeMap(() => this.deviceFromServer))
-  ).pipe(shareReplay(1));
+  private devices = this.groupIdSubject.pipe(
+    combineLatestWith(this.refreshSubject),
+    mergeMap(([groupId, _]) => {
+      if (groupId === null) {
+        return this.allDeviceFromServer;
```

Comment:
does refresh actually work for non-group scene?

Reply:
I guess "non-group scene" means the view with all devices. 
Refresh works for the case.